enhancements/multi-arch/heterogeneous-architecture.md

```diff
@@ -0,0 +1,337 @@
+- Upgrade support - CVO/Cincinnati support (heterogeneous -> heterogeneous payload upgrades only).
+- Does not include support for:
+  - image registry - internal image registry will not support manifest listed images.
+  - imagestreams importing full manifestlist content (will continue to import a single image when referencing manifestlists).
+
+Phase 2: Full basic functionality
+- Imagestream manifestlist support.
+- OLM/Operator ecosystem improvements.
+- oc debug (and other commands) support for compute architecture.
+- support toolbox.
+- support for Power and Z architectures.
+- Hypershift awareness.
+
+Phase 3: Quality of life improvements
+- First class install support - ability to create non-control-plane-architecture workers at install time and generate machineconfig
+and machineset manifests for the same.
+- Improve console's content filtering by making it architecture aware.
+- Possibly reduce mirroring costs by looking into things like sparse manifestlist.
+- Upgrade support for homogeneous payload -> heterogeneous payload and vice versa.
+
+### User Stories
+
+- As an enterprise customer looking to save compute resource cost, I want to be able to run part of my application workload on ARM instead of x86.
+- As the owner of an application that runs partially on x86 and partially on ARM, I want to be able to run the entire application within a
+single cluster so I do not need to pay the cost of higher latency between clusters, or worry about the HA-resilience of cross-cluster bridges.
+
+### API Extensions
+
+Upgrades to a heterogeneous payload will require an additional field either in the Infrastructure or the ClusterVersion resource to indicate that
+the user wants to upgrade to a heterogeneous payload.
+
+### Implementation Details/Notes/Constraints
+
+Operators supporting multiple architectures cannot stop supporting a given architecture since it would break heterogeneous clusters which are
+trying to upgrade.
+
+### Risks and Mitigations
+
+Heterogeneous payloads will only ship if the corresponding homogeneous payloads for all architectures can ship and are not broken due to a bug in a
+specific architecture.
+
+## Design Details
+
+### RHCOS Support for ARM Azure
+
+The RHCOS pipeline will need to generate Azure disk images for ARM. However as of today, RHEL does not support hyper-v for ARM and that support is
+landing in RHEL8.6. Hence 4.11 builds of RHCOS for ARM Azure will be based on RHEL8.6. It is TBD whether a separate pipeline with early access RHCOS
+builds will be needed as RHEL8.6 would not have GA'ed before 4.11.
+
+### OCP Build & Release
+
+Phase 0 will involve building and publishing of [nightly builds][OCPPLAN-7640]. The release payload will be a manifest listed image with the component
+images being manifest listed as well. All 4 architectures - x86, ARM, Power and Z will be included in the manifestlist. Existing architecture specific
```

Comment:
```suggestion
images being manifestlist references as well. All 4 architectures - x86, ARM, Power and Z will be included in the manifestlist. Existing architecture specific
```

Reply:
here it says all 4 arches will be included in phase 0, but earlier phase 0 was described as only having ARM+x86.